### PR TITLE
Support for PHP 7.4 and PHP 8.0

### PR DESCRIPTION
**Pull request by frost-byte**
There is a lot to cover in this PR.

Here is a list of just some of the changes:

- Add support for PHP7.4 and PHP 8.0
- Use bionic instead of xenial for all Docker Images
    - Provides newer version of ssl1.1 out of the box  
- Keep support for 7.3, but remove previous versions < 7.4
- Miscellaneous Changes
    - Update tests to use latest version of phpunit (v.9.5.4)
    - Update extensions to newer/more recent builds
    - Added a function for using latest commit from git for building extensions
        - imagick
        - phalcon
        - pq
        - raphf
        -  stackdriver_debugger
        - v8js
- Libraries
    - Add librabbitmq to support newer version of the amqp extension
    - Cassandra uses libuv provided by Datastax
    - Removed libvips, can now use version bundled with bionic (libvips42)
- Extensions that are now built into PHP natively and no longer bundled as separate extensions
    - apcu
    - jsond/jsonc (7.4, 8.0 are built in)
    - sodium
- Extensions that are obsolete
    - apcu_bc (7.4, 8.0 are built in)
- Extensions that require updates upstream for php8
    - hprose
    - lua
    - opencensus (official compatible release should be coming soon)
    - phalcon
    - stackdriver_debugger (I'm testing/updating a fork to support php8)
    - tcpwrap
- Extensions not building
    - v8js - `could not find libv8_libplatform library`

There are a few things that still need to be reviewed:
- Re-enable license checks in container structure testing (testapps containers)
- Run End-to-End tests
- The previous nginx configuration/version used was no longer recognizing the `set_by_lua` directive, I have switched to using the ppa from Ondrej for nginx-mainline v1.19; if I can determine enable the lua module for the package in bionic, then that can be used in its place

I have included a NOTES.md file containing some of the steps required to build packages, create distributions and build images.
Some of this information is spread about in the previous release, but some was not clear.

Additional customizations have also been added, so you can specify your own GCS buckets and use your GCP Project ID to host the containers and package artifacts for the distributions you create for each PHP version supported.

### CODE REVIEW

File: check-versions/.phpunit.result.cache [hunk 1]
```diff
@@ -0,0 +1 @@
+C:37:"PHPUnit\Runner\DefaultTestResultCache":302:{a:2:{s:7:"defects";a:3:{s:29:"VersionTest::testPHP73Version";i:3;s:29:"VersionTest::testPHP74Version";i:3;s:29:"VersionTest::testPHP80Version";i:3;}s:5:"times";a:3:{s:29:"VersionTest::testPHP73Version";d:0.105;s:29:"VersionTest::testPHP74Version";d:0.076;s:29:"VersionTest::testPHP80Version";d:0.075;}}}
```
Comment (paumas) [inline check-versions/.phpunit.result.cache:1]: I think this file not require. Add this file to  .gitignore

File: php-base/build-scripts/dump_php_versions.sh [hunk 2]
```diff
@@ -18,7 +18,8 @@
 # A shell script for dumping php versions to files.
 set -xe
 
-PHP_SHORT_NAMES=(php71 php72 php73 php74)
+PHP_SHORT_NAMES=(php73 php74 php80)
+# PHP_SHORT_NAMES=(php73 php74 php80)
```
Comment (paumas) [inline php-base/build-scripts/dump_php_versions.sh:22]: Remove this line

File: testapps/php80_custom/tests/.phpunit.result.cache [hunk 3]
```diff
@@ -0,0 +1 @@
+C:37:"PHPUnit\Runner\DefaultTestResultCache":697:{a:2:{s:7:"defects";a:4:{s:50:"Google\Cloud\tests\PHP80CustomTest::testExtensions";i:3;s:74:"Google\Cloud\tests\PHP80CustomTest::testApcIsAbleToExecuteCommonOperations";i:4;s:54:"Google\Cloud\tests\PHP80CustomTest::testImagickCanLoad";i:4;s:62:"Google\Cloud\tests\PHP80CustomTest::testFrontControllerFileEnv";i:4;}s:5:"times";a:4:{s:50:"Google\Cloud\tests\PHP80CustomTest::testExtensions";d:0.033000000000000002;s:74:"Google\Cloud\tests\PHP80CustomTest::testApcIsAbleToExecuteCommonOperations";d:0.10199999999999999;s:54:"Google\Cloud\tests\PHP80CustomTest::testImagickCanLoad";d:0.0080000000000000002;s:62:"Google\Cloud\tests\PHP80CustomTest::testFrontControllerFileEnv";d:0.0030000000000000001;}}}
```
Comment (paumas) [inline testapps/php80_custom/tests/.phpunit.result.cache:1]: Add this file to .gitignore

File: check-versions/.phpunit.result.cache (hunk 1 above)
Comment (frost-byte) [reply check-versions/.phpunit.result.cache:1]: Thanks I'll be updating this shortly, just have to disable stackdriver for php8, as it still needs some work for it to be compatible. The other two issues will be updated with the next submit.

**Comment by frost-byte** [comment]
I'm going to fix issues with the repo caused by removing the .cache file from the previous commits.

**Comment by frost-byte** [comment]
Going to close this and resubmit it...

**Pull request closed without merging**